osu.Game/Skinning/LegacySkin.cs

```diff
@@ -124,6 +124,9 @@ protected override void Dispose(bool isDisposing)
                     {
                         case LegacySkinConfiguration.LegacySetting.Version:
                             return SkinUtils.As<TValue>(new Bindable<decimal>(Configuration.LegacyVersion ?? LegacySkinConfiguration.LATEST_VERSION));
+
+                        case LegacySkinConfiguration.LegacySetting.LayeredHitSounds:
+                            return SkinUtils.As<TValue>(new Bindable<bool>(Configuration.LayeredHitSounds ?? true));
```

Comment:
This is actually redundant, to be precise on what you should do, `LegacySkinDecoder` simply takes all lines that're not handled specially as keys and values added to `LegacySkinConfiguration.ConfigDictionary`, so you would only need the enum member `LayeredHitSounds` to retrieve the value from the dictionary with, i.e. remove the property from `LegacySkinConfiguration` and remove the case added in `LegacySkinDecoder`.

Reply:
I see, thank you for your explanation I definitely misunderstood how the `SkinConfiguration` class works. I removed the property from `LegacySkinConfiguration` as you suggested and now access the property by string:
`GetConfig<string, bool>("LayeredHitSounds")`
Is that the proper way to do it?

Reply:
Almost there! the lookup should've been `LegacySetting.LayeredHitSounds` as the name of the enum member `LayeredHitSounds` will then be converted to a string to use as the lookup key in the dictionary, hence why the lookup type is almost always an enum, that's because it makes life easier :P, we definitely don't want to search what was the name of a config value lookup everytime we want to retrieve one.

Reply:
That's what I thought at first but then in `GetConfig` there is no default case for `LegacySkinConfiguration.LegacySetting` meaning it only tests for `LegacySkinConfiguration.LegacySetting.Version` and breaks, thus returning null when trying to retrieve the `LayeredHitSounds` property.

Maybe the case in `GetConfig` should be kept or a default case that translates the enum to a string could be added. However that would be exactly what the default case does when it is none of the enums that have specific cases thus inducing code redundancy.

Reply:
sounds like the version case should be:
```csharp
case LegacySkinConfiguration.LegacySetting legacySetting when legacySetting == LegacySkinConfiguration.LegacySetting.Version:
    // retrieve version.
```